src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoderBasicLatin.cs

```diff
@@ -135,9 +135,21 @@ public override unsafe int FindFirstCharacterToEncode(char* text, int textLength
                     // Load the next 16 characters, combine them to one byte vector.
                     // Chars that don't cleanly convert to ASCII bytes will get converted (saturated) to
                     // somewhere in the range [0x7F, 0xFF], which the NeedsEscaping method will detect.
-                    Vector128<sbyte> sourceValue = Sse2.PackSignedSaturate(
-                        Sse2.LoadVector128(ptr),
-                        Sse2.LoadVector128(ptr + Vector128<short>.Count));
+                    Vector128<sbyte> sourceValue;
+
+                    if (Sse2.IsSupported)
+                    {
+                        sourceValue = Sse2.PackSignedSaturate(
+                            Sse2.LoadVector128(ptr),
+                            Sse2.LoadVector128(ptr + Vector128<short>.Count));
+                    }
+                    else
+                    {
+                        Debug.Assert(AdvSimd.Arm64.IsSupported);
+
+                        Vector64<sbyte> tmp = AdvSimd.ExtractNarrowingSaturateLower(AdvSimd.LoadVector128(ptr));
```

Comment:
Here and in line 193, you create a temporary `Vector64` for the lower portion, and then you pass it to the "upper" method. Consider naming these temp variables `lower`.